### DISCUSSION TITLE
Detekt's 2.0 cli

### DISCUSSION THREAD

**Discussion by BraisGabin** [detekt/detekt#3616] [Ideas]
The same as #2680 but for talk about breaking changes that we could like to have in the cli once Dekte 2.0 arrives. The same as before, there's nothing wrong in 1.0. We don't need a breaking change in the inmediate future but it's good to talk about ideas.

My ideas:

I think that we should simplify our config file. It should only contain the configuration of the rules. And maybe the `config` about the exclusion... But I'm not sure about that last part. 

The other things could be moved out. For example `maxIssues` by default is 0. To change it we could use a flag like `--max-issues=10` or `--max-issues=-1`

`console-reports` doesn't seems like a good thing. If they are reports we should allow the user to have them in other formats to process them later. And if they are not "really" reports we should hide them behind a `verbose` mode.

And the processors that we support right now are two types: show progress and show counts. I think that the counts should be displayed in some reports and print them in the `versbose` mode. And the progress the same, behind the `verbose` mode or drop it. Or maybe we could rethink it and print the name of the file that we are processing. Somthing like `tar` works with `-v`.

And now that I'm thinking it... probably we should move the configuration file outside the `cli` because we want to use the same configuration file for the `gradle-plugin`. Same thing for the `baseline`.

**Comment by arturbosch** [comment]
https://github.com/detekt/detekt/issues/2680#issuecomment-657709706

I agree with moving out the build and processor, console report stuff out of the config/hiding it behind `--verbose`.
Normal users do not need to know about processors, listeners and extension IDs.

Introducing a `--max-issues` flag would be non-breaking and will be supported with the tooling-api (https://github.com/detekt/detekt/pull/2861/files#diff-2aa2bc358229c132fb7a02ff5a800b3cR12).

The defaut config and the baseline feature is already part of core :).

**Comment by arturbosch** [comment]
from https://github.com/detekt/detekt/issues/2267#issuecomment-766433748

remove --fail-fast
activate --build-upon-default-config by default (remove flag then)
